generate.py

```diff
@@ -57,29 +59,47 @@ def main():
         residual_channels=wavenet_params['residual_channels'],
         dilation_channels=wavenet_params['dilation_channels'],
         quantization_channels=wavenet_params['quantization_channels'],
-        use_biases=wavenet_params['use_biases'])
+        use_biases=wavenet_params['use_biases'],
+        fast_generation=args.fast_generation)
 
     samples = tf.placeholder(tf.int32)
 
     next_sample = net.predict_proba(samples)
-
-    saver = tf.train.Saver()
+    
+    if args.fast_generation:
+        sess.run(tf.initialize_all_variables())
```

Comment:
@mecab I'm not sure what you mean. Let me describe what we are trying to do and make sure we are on the same page. We create some new variables for the generation: `state_buffer` and `pointer` for each queue. We initialize them, and then initialize all the other variables from a checkpoint using the `saver` (see the code in your comment below).

I think this achieves that well enough. It could be cleaner though...